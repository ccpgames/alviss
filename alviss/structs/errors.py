__all__ = [
    'AlvissError',
    'AlvissStubberError',
    'AlvissStubberSyntaxError',
    'AlvissStubberInvalidTypeName',
]


class AlvissError(Exception):
    pass


class AlvissStubberError(AlvissError):
    pass


class AlvissStubberSyntaxError(AlvissStubberError, ValueError):
    def __init__(self, message: str, field_name: str = '?'):
        super().__init__(message)
        self.field_name = field_name

    def __str__(self):
        return f'{super().__str__()} (field_name="{self.field_name}")'


class AlvissStubberInvalidTypeName(AlvissStubberSyntaxError, TypeError):
    def __init__(self, message: str, field_name: str = '?', type_name: str = '?'):
        super().__init__(message, field_name)
        self.type_name = type_name

    def __str__(self):
        return f'{super().__str__()} (type_name="{self.type_name}", field_name="{self.field_name}")'
