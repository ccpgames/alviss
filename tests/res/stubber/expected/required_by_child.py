__all__ = [
    'CfgTopSecondShouldBeOptionalStub',
    'CfgTopSecondMustExistStub',
    'CfgTopSecondStub',
    'CfgTopStub',
    'CfgStub',
    'AlvissConfigStub',
]

from typing import *
from alviss.structs import Empty
from alviss.structs.cfgstub import _BaseCfgStub
from alviss.structs import BaseConfig


class CfgTopSecondShouldBeOptionalStub(_BaseCfgStub, dict):
    bar: Union[str, Empty]


class CfgTopSecondMustExistStub(_BaseCfgStub, dict):
    foo: str


class CfgTopSecondStub(_BaseCfgStub, dict):
    should_be_optional: Union[CfgTopSecondShouldBeOptionalStub, Empty]
    must_exist: CfgTopSecondMustExistStub


class CfgTopStub(_BaseCfgStub, dict):
    second: CfgTopSecondStub


class CfgStub(_BaseCfgStub, dict):
    top: CfgTopStub


class AlvissConfigStub(BaseConfig, CfgStub):
    pass
