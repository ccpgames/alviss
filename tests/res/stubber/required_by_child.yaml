top:  # Required
  second:  # Required
    should_be_optional:  # NOT Required
      bar: str  # NOT Required
    must_exist:  # Required
      foo: str*  # Since this is required, all its parents must also be!