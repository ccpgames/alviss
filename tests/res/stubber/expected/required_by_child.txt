__all__ = [
    'CfgTopSecondShouldBeOptionalStub',
    'CfgTopSecondMustExistStub',
    'CfgTopSecondStub',
    'CfgTopStub',
    'CfgStub',
    'AlvissConfigStub',
]

from typing import *
from alviss.structs import Empty
from alviss.structs.cfgstub import _BaseCfgStub
from alviss.structs import BaseConfig


class CfgTopSecondShouldBeOptionalStub(_BaseCfgStub):
    bar: Union[str, Empty]


class CfgTopSecondMustExistStub(_BaseCfgStub):
    foo: str


class CfgTopSecondStub(_BaseCfgStub):
    should_be_optional: Union[CfgTopSecondShouldBeOptionalStub, Empty]
    must_exist: CfgTopSecondMustExistStub


class CfgTopStub(_BaseCfgStub):
    second: CfgTopSecondStub


class CfgStub(_BaseCfgStub):
    top: CfgTopStub


class AlvissConfigStub(BaseConfig, CfgStub):
    pass
